### PR TITLE
[mobile] Click the button in the dialog sample closes

### PR DESCRIPTION
**Pull request by DohyungLim**
- The dialog disappears when the button is clicked.

### CODE REVIEW

File: examples/mobile/UIComponents/components/dialogs/dialog.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+(function () {
+	var page = document.getElementById("demo-dialog-page"),
+		buttons = document.querySelectorAll(".ui-btn"),
+		idx;
+
+	page.addEventListener("pageshow", function () {
+		for (idx = 0; idx < buttons.length; idx++) {
+			buttons[idx].addEventListener("vclick", function () {
+				window.history.back();
```
Comment (TomaszLukawskiSam) [inline examples/mobile/UIComponents/components/dialogs/dialog.js]: The `tau.history.back()` is preferred.

File: examples/mobile/UIComponents/components/dialogs/dialog.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+(function () {
+	var page = document.getElementById("demo-dialog-page"),
+		buttons = document.querySelectorAll(".ui-btn"),
+		idx;
+
+	page.addEventListener("pageshow", function () {
```
Comment (TomaszLukawskiSam) [inline examples/mobile/UIComponents/components/dialogs/dialog.js]: You have to take account remove this listener...
Now, this listener will be add many times.

You can do similar like `examples/mobile/UIComponents/components/content/searchinput/searchinput.js`

**Review by TomaszLukawskiSam — CHANGES_REQUESTED** [review]

File: examples/mobile/UIComponents/components/dialogs/dialog.js (hunk 2 above)
Comment (DohyungLim) [reply examples/mobile/UIComponents/components/dialogs/dialog.js]: I remove addEventListener.
and only call tau.history.back() in for loop.

done.

File: examples/mobile/UIComponents/components/dialogs/dialog.js (hunk 1 above)
Comment (DohyungLim) [reply examples/mobile/UIComponents/components/dialogs/dialog.js]: ok. I agree.

done.

File: examples/mobile/UIComponents/components/dialogs/dialog.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+(function (document, tau) {
+	var buttons = document.querySelectorAll(".ui-btn"),
+		idx;
+
+	for (idx = 0; idx < buttons.length; idx++) {
+		buttons[idx].addEventListener("vclick", function () {
+			tau.history.back();
+		})
+	}
+}());
```
Comment (TomaszLukawskiSam) [inline examples/mobile/UIComponents/components/dialogs/dialog.js:24]:
```
(function () {
	var page = document.getElementById("demo-dialog-page"),
		buttons = document.querySelectorAll(".ui-btn"),
		idx;

	function onClick() {
		tau.history.back();
	}

	function onPageShow() {
		for (idx = 0; idx < buttons.length; idx++) {
			buttons[idx].addEventListener("vclick", onClick);
		}
	}	

	function onPageHide() {
		for (idx = 0; idx < buttons.length; idx++) {
			buttons[idx].removeEventListener("vclick", onClick);
		}
	}

	page.addEventListener("pageshow", onPageShow);
	page.addEventListener("pagehide", onPageHide);
}());
```

Comment (DohyungLim) [reply examples/mobile/UIComponents/components/dialogs/dialog.js:24]: done.

**Review by TomaszLukawskiSam — APPROVED** [review]

**Comment by TomaszLukawskiSam** [comment]
Looks good to me.

File: examples/mobile/UIComponents/components/dialogs/dialog.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+	}
+
+	function onPageHide() {
+		for (idx = 0; idx < buttons.length; idx++) {
+			buttons[idx].removeEventListener("vclick", onClick);
+		}
+	}
+
+	page.addEventListener("pageshow", onPageShow);
+	page.addEventListener("pagehide", onPageHide);
```
Comment (HunseopJeong) [inline examples/mobile/UIComponents/components/dialogs/dialog.js]: "pagebeforeshow" and "pagebeforehide" is commonly used in TAU.

Comment (TomaszLukawskiSam) [reply examples/mobile/UIComponents/components/dialogs/dialog.js]: You are right.

**Review by HunseopJeong — APPROVED** [review]
LGTM

**Pull request merged**